improver_tests/utilities/test_cube_checker.py

```diff
@@ -217,37 +237,50 @@ def test_mismatch_in_first_cube(self):
     def test_mismatch_in_second_cube(self):
         """Test when finding a one-way mismatch, so that the first cube has
         a missing coordinate. This returns a list with the missing coordinate
-        name.l"""
-        cube = set_up_cube()
-        first_cube = cube.copy()
-        first_cube.remove_coord("time")
-        second_cube = cube.copy()
+        name."""
+        first_cube = self.cube.copy()
+        second_cube = self.cube.copy()
+        first_cube.remove_coord("realization")
         result = find_dimension_coordinate_mismatch(
             first_cube, second_cube, two_way_mismatch=False
         )
         self.assertIsInstance(result, list)
-        self.assertListEqual(result, ["time"])
+        self.assertListEqual(result, ["realization"])
 
     def test_two_way_mismatch(self):
         """Test when finding a two-way mismatch, when the first and second
         cube contain different coordinates."""
-        cube = set_up_cube()
-        first_cube = cube.copy()
-        first_cube.remove_coord("time")
-        second_cube = cube.copy()
+        first_cube = self.cube.copy()
+        second_cube = self.cube.copy()
         second_cube.remove_coord("realization")
+        second_cube = add_coordinate(second_cube, [10, 20], "height", "m")
+        iris.util.promote_aux_coord_to_dim_coord(second_cube, "height")
         result = find_dimension_coordinate_mismatch(first_cube, second_cube)
         self.assertIsInstance(result, list)
-        self.assertListEqual(result, ["time", "realization"])
+        self.assertListEqual(result, ["height", "realization"])
 
 
 class Test_spatial_coords_match(IrisTest):
     """Test for function testing cube spatial coords."""
 
     def setUp(self):
         """Create two unmatching cubes for spatial comparison."""
-        self.cube_a = set_up_cube(num_grid_points=16)
-        self.cube_b = set_up_cube(num_grid_points=10)
+        data_a = np.ones((1, 16, 16), dtype=np.float32)
+        data_a[:, 7, 7] = 0.0
+        data_b = np.ones((1, 10, 10), dtype=np.float32)
+        data_b[:, 7, 7]
```

Comment:
Codacy found an issue: [Statement seems to have no effect](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5942293)